### PR TITLE
MB-63643: Fix missing num_threads clauses

### PR DESCRIPTION
**Pull request by CascadingRadium**
- In some places the #pragma omp statements were missing the num_threads clause, 
leading to the global OMP Config being ignored here

### CODE REVIEW

**Review by abhinavdangeti — APPROVED** [review]

**Comment by abhinavdangeti** [comment]
@CascadingRadium please wait for a review from @deepkaran as well.

File: faiss/utils/distances.cpp
Comment (deepkaran) [inline faiss/utils/distances.cpp:149]: There is another function exhaustive_L2sqr_seq in this file with similar code. It would be good to change that as well while we are at it.

Comment (abhinavdangeti) [reply faiss/utils/distances.cpp:149]: Done.

**Review by abhinavdangeti — APPROVED** [review]

**Pull request merged**
